content/blog/2022-07-28-switching-to-dvc-from-jupyter-vscode.md

```diff
@@ -0,0 +1,154 @@
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/syncing-data-to-aws-s3/1192
+tags:
+  - MLOps
+  - DVC
+  - Git
+  - VSCode
+  - Juptyer Notebooks
+---
+
+For many machine learning engineers, the starting point of a project is a
+Jupyter notebook. This is fine for running a few experiments, but there comes a
+point where you need to scale the project to accomodate hundreds or even
+thousands more experiments. It will also be important to track the experiments
+you run so that when you find an exceptional model, you'll be able to reproduce
+it and get it ready for production.
+
+In this tutorial, we're going to start a project with a Juptyer notebook in
+VSCode. Then we'll convert it to DVC to make reproducible experiments and use
+the DVC VSCode extension to run new experiments and see how to compare them all.
+[Here's the project](https://github.com/iterative/stale-model-example/tree/jupyter-to-dvc)
+we'll be working with.
+
+## Start training with the notebook
+
+Many times you'll start an ML project with a few cells in a notebook just to
+test out some thoughts you might have. So you might have a simple notebook where
+you set some hyperparameters, load your data, train a model and evaluate its
+metrics, and then save the model. That's what we're doing in the
+`bicycle_experiments.ipynb` file.
+
+![Jupyter notebook cells](/uploads/images/2022-07-28/jupyter-notebook.png)
+
+We have all of the cells in place so we can start running experiments. This is
+usually fine for training models for a while. Although there will eventually be
+a point where you are powering through experiments for the day and you want to
+compare metrics across experiments. You might also end up with a great model,
+but you have no idea what code you used or which data was used to train this
+model.
+
+That makes reproducing the experiment impossible and you're left with a great
+model you may not be able to use in production. Once you reach the point where
+you are trying to reproduce models or compare metrics from multiple experiments,
+it might make sense to look at a data versioning and model experiment tracking
+tool like DVC.
+
+## Refactor the Jupyter notebook to use DVC
+
+We're going to take the existing Jupyter notebook and break the cells out into
+files and stages that DVC tracks for you. First, we'll create a `train.py` file
+to handle the model training stage of the experiment. This file will have the
+`Get params`, `Load training data`, `Train model`, and `Save model` cells from
+the earlier notebook.
+
+Next, we'll make a `evaluate.py` file that will take a saved model and get the
+metrics for how well it performs. This file will have the `Set test variables`,
+`Load model and test data`, `Get model predictions`,
+`Calculate model performance metrics`, and `Save model performance metrics`
+notebook cells.
+
+That's it! Now you have all of the steps that you executed in your Jupyter
```

Comment:
What do you think about making this step not sound so simple? IMO this will often be not as easy as it sounds here since the notebook is unlikely to be as well-structured as your example 😄 . Is it better to acknowledge the challenge of this part and explain what's important? People may need to reorder their code, add functions, etc. to make a workable script, but the payoff is that you now have a reproducible way to run your model training.